eslintPlugin/src/noAnimatedStyleToNonAnimatedComponent.ts

```diff
@@ -0,0 +1,155 @@
+        if (node.name.type === AST_NODE_TYPES.JSXNamespacedName) return; // XML-based namespace syntax: <Animated:View>
+
+        const sourceCode = context.getSourceCode();
+        const tokensBefore = sourceCode.getTokensBefore(node);
+        const componentName = node?.name?.name;
+
+        function isVariableDefinedAs(
+          variableName: string,
+          expectedToken: string
+        ) {
+          let isAnimated = false;
+          const variableNameTokenIds: Array<number> = [];
+
+          tokensBefore.forEach((token, idx) => {
+            if (token.value === variableName) {
+              variableNameTokenIds.push(idx);
+            }
+          });
+
+          variableNameTokenIds.forEach((idx) => {
+            if (tokensBefore[idx + 2].value === expectedToken) {
+              isAnimated = true;
+            }
+          });
+          return isAnimated;
+        }
+
+        function checkIdentifierNode(styleExpression: TSESTree.Identifier) {
+          const variableName = styleExpression.name;
+
+          const isAnimatedStyle = isVariableDefinedAs(
+            variableName,
+            'useAnimatedStyle'
+          );
+
+          if (isAnimatedStyle) {
+            context.report({
+              node,
+              messageId: 'animatedStyle',
+              data: { componentName, variableName },
+            });
+          }
+        }
+
+        function checkObjectNode(styleExpression: TSESTree.ObjectExpression) {
+          const properties = styleExpression.properties;
+          properties.forEach((property) => {
+            if (property.type === AST_NODE_TYPES.SpreadElement) {
+              return; //Ignore spread elements
+            }
+
+            if (property.value.type === 'Identifier') {
+              const variableName = property.value.name;
+              if (isVariableDefinedAs(variableName, 'useSharedValue')) {
+                const propertyName =
+                  'name' in property.key ? property.key.name : variableName;
+                context.report({
+                  node,
+                  messageId: 'sharedValue',
+                  data: {
+                    propertyName,
+                    propertyValue: variableName,
+                    componentName,
+                  },
+                });
+              }
+            }
+          });
+        }
+
+        function checkArrayNode(styleExpression: TSESTree.ArrayExpression) {
+          const arrayNodes = styleExpression.elements;
+          arrayNodes.forEach((node) => {
+            if (node?.type === 'Identifier') {
+              checkIdentifierNode(node);
+            } else if (node?.type === 'ArrayExpression') {
+              checkArrayNode(node);
+            } else if (node?.type === 'ObjectExpression') {
+              checkObjectNode(node);
+            }
+          });
+        }
+
+        if (
+          isVariableDefinedAs(componentName, 'Animated') ||
```

Comment:
People tend to do:

`import Animated as Reanimated` or so, especially in React Native .Animated & Reanimated brownfields. Will it work in this case?